doc/streams-dsl.md

```diff
@@ -0,0 +1,78 @@
+# Kafka Streams DSL
```

Comment:
This duplicates much of what is already in the Word Count example. I'm wondering if we should improve the [README](https://github.com/FundingCircle/jackdaw/blob/master/examples/word-count/README.md) and associated [example](https://github.com/FundingCircle/jackdaw/blob/master/examples/word-count/src/word_count.clj) instead.